### PR TITLE
Inline Angular async tests

### LINKED ISSUE #3791: Ugly formatting for Angular unit tests
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAmHwBU4BnGAITgBMAafAQzIE8ox8BffAMxwnT4A5AAFGUAOYBXADaMcAekg44C+BVSShAbgA6UDNjyF8AQUyYAwgOxQEBbnwHCAdAsYWXkLNHs79+jTkYDioAEZwABRC5lY2vrBCDJEAlPgAvAB8hPr4+BE8uHAAooxgABaRzGxgkakZ2cC5eSTkVLRe0Dyo0iqkGpIAshA0slFNUC0tQWByOIwwqNBkSPgA2rHWPnawALrNeZwpnVioMnBbtvZkqXqTXCkpd3moMNFk5RCyNPghcAtwfAwcqAjyYJJMVjsOppLI5e55SBQCi8VAADxgUhUGVaFGoNC8KgBlwSb028R2MCeB1+ywIYJx3QxWLgLiCYSkEmK53Q9hOV1gAElkTBxGA4M8WnA0Zg4GA3mDjjAINRiDgpMCWLdmkdqfdXpEAAYfL4yH7lRgAN1BZCBr3OwjBQkNDGq0PqcImUyRKKZmOx6Vx7QJf2JFPskXJ23seu9dKYFkZ6P9rPZnO5cF5sH5pOFFDFEpp0tl8qqnkWMHOSogxQAjlJGDJooMWPgAGKoHAo0ySWTyMwWISxh7Dg26EAm774FRQII4O2VwGaJj4coARiBjAk49dUNqHsaNJ9BD9LJx-WDhP+8BJlMjFlvMcleVPKjZcHg8ssFsk5G1CNpEVaVOc4fkDV9UzgDkuR5PkoAWVBrQzLMYGffBizlN5vEwM4OnrOAcBYABlOBznlXBonXIcXHgDFrFgGMaIgejRU0ZtWw7LsCB7aQ5gHcFh11O4jjuEA6BACBMEWZZkFATBQlgAB1VAaGBZAAA4AAZxNFMJlNU8pkAAJnEshNAkc4AEUpAgeBkB4RsyDgHT5jOcytnQRhkBAKBfDEkAwnmMAAGsPyIzAynM5AYHVZyQAAKzINFKCC0KYCIxheQAGU0OB7McuKIq7AjvLCRgwhYGRoH8pz0FQaLYvEqQnOIcqyHymQnPEzQnLwAAFeYJE8jqupAFR607OABq3YakAczrCv4JzFPmTBvPk8gCOtfz5H4AB3Pr5AQdqUH8mgIDAEa4uYGArvEr4YEwDUjKuzhOCAA)
```sh
--arrow-parens 
```

**Input:**
```jsx
import { TestBed, async } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent]
    }).compileComponents();
  }));
  it('should create the app', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  }));
  it(`should have as title 'app'`, async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('My First Angular App');
  }));
  it("should render title in a h1 tag", async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('My First Angular App');
  }));
});

```

**Output:**
```jsx
import { TestBed, async } from "@angular/core/testing";
import { AppComponent } from "./app.component";

describe("AppComponent", () => {
  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [AppComponent]
      }).compileComponents();
    })
  );
  it(
    "should create the app",
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement.componentInstance;
      expect(app).toBeTruthy();
    })
  );
  it(
    `should have as title 'app'`,
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement.componentInstance;
      expect(app.title).toEqual("My First Angular App");
    })
  );
  it(
    "should render title in a h1 tag",
    async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      fixture.detectChanges();
      const compiled = fixture.debugElement.nativeElement;
      expect(compiled.querySelector("h1").textContent).toContain(
        "My First Angular App"
      );
    })
  );
});

```

**Expected behavior:**
I would expect functions wrapped in Angular's async wrapper to be formatted the same way that native async arrow functions are in unit tests.

In this case, the test names and body should be on one line.

### PR DESCRIPTION
**Pull request by duailibe**
Closes #3791

Might be better to review ignoring whitespace https://github.com/prettier/prettier/pull/4010/files?w=1

### CODE REVIEW

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -5211,22 +5211,36 @@ function isObjectType(n) {
...
+    (n.callee.type === "Identifier" &&
+      testSetupRe.test(n.callee.name) &&
+      n.arguments.length === 1 &&
+      (((n.arguments[0].type === "FunctionExpression" ||
+        n.arguments[0].type === "ArrowFunctionExpression") &&
+        n.arguments[0].params.length === 0) ||
+        (n.arguments[0].type === "CallExpression" &&
+          n.arguments[0].callee.type === "Identifier" &&
+          n.arguments[0].callee.name === "async")))
   );
```
Comment (j-f1) [inline src/language-js/printer-estree.js]: What do you think about splitting this 30ish-line expression into multiple `if` statements?

**Comment by duailibe** [comment]
Yeah it only fixes part of the problem

File: tests/test_declarations/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -25,13 +21,11 @@ it("should create the app", async(() => {
...
 
-it(\`should have as title 'app'\`, async(() => {
-  // code
-}));
-
-it("should render title in a h1 tag", async(() => {
-  // code
-}));
+it("does something really long and complicated so I have to write a very long name for the test", async(
+  () => {
```
Comment (suchipi) [inline tests/test_declarations/__snapshots__/jsfmt.spec.js.snap]: I think this function expression should hug the `async` call expression in order to be consistent with the other inlined test patterns.

Comment (suchipi) [reply tests/test_declarations/__snapshots__/jsfmt.spec.js.snap]: Oh that's probably what you meant by it only fixing part of the problem

**Comment by duailibe** [comment]
Not gonna work on this right now, if anyone wants to take over

**Pull request closed without merging**